Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -3149,8 +3148,11 @@ ExceptionOr<void> WebGLRenderingContextBase::texImageSourceHelper(TexImageFuncti
     if (isContextLost())
         return { };
 
-    return std::visit([this, functionID, target, level, internalformat, border, format, type, xoffset, yoffset, zoffset, inputSourceImageRect, depth, unpackImageHeight](auto&& source) {
-        return texImageSource(functionID, target, level, internalformat, border, format, type, xoffset, yoffset, zoffset, inputSourceImageRect, depth, unpackImageHeight, *source);
+    return std::visit([weakThis = createRefForContextObject(), functionID, target, level, internalformat, border, format, type, xoffset, yoffset, zoffset, inputSourceImageRect, depth, unpackImageHeight](auto&& source) {
```

Comment:
Just capture `this, protectedThis = Ref { *this }` here.

There is no need for a weak ptr, the lambda runs synchronously.